instance-scheduler/main_test.go

```diff
@@ -1,29 +1,17 @@
 package main
 
 import (
-	"fmt"
 	"github.com/aws/aws-lambda-go/events"
-	"net/http"
-	"net/http/httptest"
-	"os"
 	"testing"
 )
 
 func TestHandler(t *testing.T) {
-	os.Setenv("INSTANCE_SCHEDULING_ACTION", "Test")
-
-	t.Run("Successful Request", func(t *testing.T) {
-		ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-			w.WriteHeader(200)
-			fmt.Fprintf(w, "Testing")
-		}))
-		defer ts.Close()
-
-		DefaultHTTPGetAddress = ts.URL
-
+	t.Run("Testing the lambda function", func(t *testing.T) {
 		_, err := handler(events.APIGatewayProxyRequest{})
 		if err != nil {
-			t.Fatal("Everything should be ok")
+			t.Error("[ERROR]:", err)
+		} else {
+			t.Log("[INFO]:", "Test has run successfully!")
```

Comment:
Does this statement add information? If there are no assertion failures then the test will be marked as PASS anyway:

```
--- PASS: TestHandler (7.98s)
    --- PASS: TestHandler/Testing_the_lambda_function (7.98s)
PASS
ok      instance-scheduler      8.111s
```
I don't see the value of adding a statement like this.

Reply:
Fair point, although it's testing the tests logic this way, making sure it works correctly. I can remove it, but again when expanding on tests, I will be using a lot of logging explaining what has been tested with each condition. 

In this case, the test just runs the lambda function and tests its functionality, so not much to say other than it works.

Reply:
In automated tests the more common approach is to use assertion messages as a means to explain the test failure. Here is an example: https://github.com/ministryofjustice/modernisation-platform-environments/blob/3190c6dceb65e695bd49f48f00cb560f9bc7cb75/scripts/tests/go-tests/sandbox_accounts_test.go#L142
If any of the assertion fails then the testing library prints FAIL at the end to indicate that the overall test has failed. Meaning that, when the test is marked as PASS it is already implied that there were no assertion failures and the test succeeded.

Reply:
agree, which makes logging rather redundant... 

I will delete the code to reflect that, but don't read too much into this test. It is more of a starting point and getting it enabled in the workflow.